okhttp/src/jvmTest/java/okhttp3/MediaTypeJavaTest.kt

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (C) 2022 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okhttp3
+
+import java.nio.charset.StandardCharsets
+import java.util.*
+import kotlin.test.assertEquals
+import kotlin.test.assertNull
+import okhttp3.MediaType.Companion.toMediaType
+import okhttp3.internal.platform.Platform.Companion.isAndroid
+import org.junit.jupiter.api.Test
+
+class MediaTypeJavaTest: MediaTypeGetTest() {
+  override fun MediaType.charsetName(): String? = this.charset()?.name()
+
+  @Test fun testCharsetNameIsDoubleQuotedAndSingleQuotedAndroid() {
+    val mediaType = "text/plain;charset=\"'utf-8'\"".toMediaType()
+    if (isAndroid) {
+      // Charset.forName("'utf-8'") == UTF-8
+      assertEquals("UTF-8", mediaType.charsetName())
+    } else {
+      assertNull(mediaType.charset())
+    }
+  }
+
+  @Test fun testDefaultCharset() {
+    val noCharset = parse("text/plain")
+    assertEquals(
+      "UTF-8", noCharset.charset(StandardCharsets.UTF_8)!!.name()
+    )
+    assertEquals(
+      "US-ASCII", noCharset.charset(StandardCharsets.US_ASCII)!!.name()
+    )
+    val charset = parse("text/plain; charset=iso-8859-1")
+    assertEquals(
+      "ISO-8859-1", charset.charset(StandardCharsets.UTF_8)!!.name()
+    )
+    assertEquals(
+      "ISO-8859-1", charset.charset(StandardCharsets.US_ASCII)!!.name()
```

Comment:
We can replace `StandardCharsets` with `kotlin.text.Charsets` here?

Reply:
Yep - it doesn't help much, these classes are JVM+withJava(), and most of the rest of the codebase still uses StandardCharsets.  If we change, we should change in one go?

Reply:
In #6986 do the rest.